FinanceManager.UnitOfWork/Common/Abstracts/IUnitOfWorkExecuter.cs

```diff
@@ -0,0 +1,18 @@
+﻿using FinanceManager.UnitOfWork.Abstructs;
+
+namespace FinanceManager.UnitOfWork.Common.Abstracts;
+
+public interface IUnitOfWorkExecuter
```

Comment:
Looks like it should be part of IUnitOfWork itself, this way you minimize chances of type inference not working, so instead of executor.Execute<T>() you would have repository.Execute(). Also, your execution strategies should be self-sufficient (accept repository as a target) to eliminate not clear usage: specific strategies do not have specific repositories attached to them unless you follow T in Execute<T>().